src/lib/components/Post.svelte

```diff
@@ -99,18 +129,65 @@
         </Card.Footer>
     </div>
     <div class="flex flex-col items-center justify-between justify-self-end">
-        <Popover.Root>
-            <Popover.Trigger asChild let:builder>
-                <Button builders={[builder]} variant="ghost" size="icon" disabled={!post}>
-                    <div class:animate-pulse={!post}>
-                        <Share2 class="h-5 w-5" />
-                    </div>
-                </Button>
-            </Popover.Trigger>
-            <Popover.Content>
-                <h1>Not yet implemented!</h1>
-            </Popover.Content>
-        </Popover.Root>
+        <Dialog.Root bind:open>
+            <DropdownMenu.Root>
+                <DropdownMenu.Trigger asChild let:builder>
+                    <Button builders={[builder]} variant="ghost" size="icon" disabled={!post}>
+                        <div class:animate-pulse={!post}>
+                            <EllipsisVertical class="h-5 w-5" />
+                        </div>
+                    </Button>
+                </DropdownMenu.Trigger>
+                <DropdownMenu.Content class="w-56">
+                    <DropdownMenu.Group>
+                        <DropdownMenu.Item>Share</DropdownMenu.Item>
+                        <DropdownMenu.Separator />
+                        <Dialog.Trigger>Report</Dialog.Trigger>
```

Comment:
This should be red to show some importance and have a flag icon.  See how it is done in the full-page post.